From: =?UTF-8?q?Mika=20Pfl=C3=BCger?= <debian@mikapflueger.de>
Date: Sun, 4 Mar 2012 00:49:12 +0100
Subject: Perdition policy adjustments: * Allow perdition to bind to sieve
 port, read /dev/urandom, and capabilities chown and fowner * Allow
 perdition to authenticate with mysql, read directories of type
 perdition_etc_t, connect to the pop ports

---
 policy/modules/services/perdition.te |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/policy/modules/services/perdition.te b/policy/modules/services/perdition.te
index 8e92c65..9f918f4 100644
--- a/policy/modules/services/perdition.te
+++ b/policy/modules/services/perdition.te
@@ -20,6 +20,8 @@ files_pid_file(perdition_var_run_t)
 # Local policy
 #
 
+allow perdition_t self:netlink_route_socket create_netlink_socket_perms;
+allow perdition_t self:capability { chown fowner setgid setuid };
 dev_read_urand(perdition_t)
 dontaudit perdition_t self:capability sys_tty_config;
 allow perdition_t self:process signal_perms;
@@ -27,6 +29,7 @@ allow perdition_t self:tcp_socket create_stream_socket_perms;
 allow perdition_t self:udp_socket create_socket_perms;
 
 allow perdition_t perdition_etc_t:file read_file_perms;
+allow perdition_t perdition_etc_t:dir list_dir_perms;
 files_search_etc(perdition_t)
 
 manage_files_pattern(perdition_t, perdition_var_run_t, perdition_var_run_t)
@@ -46,6 +49,8 @@ corenet_tcp_sendrecv_all_ports(perdition_t)
 corenet_udp_sendrecv_all_ports(perdition_t)
 corenet_tcp_bind_generic_node(perdition_t)
 corenet_tcp_bind_pop_port(perdition_t)
+corenet_tcp_bind_sieve_port(perdition_t)
+corenet_tcp_connect_pop_port(perdition_t)
 corenet_sendrecv_pop_server_packets(perdition_t)
 
 dev_read_sysfs(perdition_t)
@@ -73,3 +78,7 @@ optional_policy(`
 optional_policy(`
 	udev_read_db(perdition_t)
 ')
+optional_policy(`
+	mysql_tcp_connect(perdition_t)
+	mysql_stream_connect(perdition_t)
+')
